packages/bpk-component-chip/src/commonTypes.ts

```diff
@@ -16,28 +16,29 @@
  * limitations under the License.
  */
 
-/* @flow strict */
-
 import PropTypes from 'prop-types';
-import { type Node } from 'react';
+import type { ComponentProps, ReactNode, SyntheticEvent } from 'react';
 
 export const CHIP_TYPES = {
   default: 'default',
   onDark: 'on-dark',
   onImage: 'on-image',
-};
+} as const;
 
-export type CommonProps = {
-  accessibilityLabel: string,
-  children: Node,
-  disabled: boolean,
-  onClick: (event: SyntheticEvent<>) => mixed,
-  selected: boolean,
-  type: $Values<typeof CHIP_TYPES>,
-  className: ?string,
-  leadingAccessoryView: ?Node,
-};
+// onClick and children are already part of the button props, but we need to specify that they are required
+interface ButtonProps extends Omit<ComponentProps<'button'>, 'type'> {
+  onClick: (event: SyntheticEvent<HTMLButtonElement>) => void | null;
+  children: ReactNode | string;
+  type?: typeof CHIP_TYPES[keyof typeof CHIP_TYPES]; // this is different from the native button type
+}
```

Comment:
How come we need to declare this interface and do this instead of just having these inside the `CommonProps`?